src/core/issues-notes.js

```diff
@@ -141,22 +141,25 @@ export function run(conf, doc, cb) {
     issueBase = conf.issueBase;
   if ($ins.length) {
     if (conf.githubAPI) {
-      ghFetch(conf.githubAPI)
-        .then(function(json) {
-          issueBase = issueBase || json.html_url + "/issues/";
-          return fetchIndex(json.issues_url, {
-            // Get back HTML content instead of markdown
-            // See: https://developer.github.com/v3/media/
-            Accept: "application/vnd.github.v3.html+json",    
-          });
-        })
-        .then(function(issues) {
-          issues.forEach(function(issue) {
-            ghIssues[issue.number] = issue;
-          });
-          handleIssues($ins, ghIssues, issueBase);
-          cb();
-        }).catch(err => {
+      const asyncFetch = async () => {
+        const json = await ghFetch(conf.githubAPI);
+        const issues = await getIssues(json);
+        issues.forEach( (issue) => {
+          ghIssues[issue.number] = issue;
+        });
+        handleIssues($ins, ghIssues, issueBase);
+        cb();
+      }
+      const getIssues = (json) => {
```

Comment:
A function expression, that is:

[const | let | var] = function () {} (or () =>
Is the creation of an anonymous function (function () {}) and the creation of a variable, and then the assignment of that anonymous function to that variable.

So the usual rules around variable hoisting within a scope -- block-scoped variables (let and const) do not hoist as undefined to the top of their block scope.

This means:

if (true) {
    doSomething() // will fail
    const doSomething = function () {}
}
Will fail since doSomething is not defined. (It will throw a  ReferenceError)

If you switch to using var you get your hoisting of the variable, but it will be initialized to undefined so that block of code above will still not work. (This will throw a TypeError since doSomething is not a function at the time you call it)

As far as standard practices go, you should always use the proper tool for the job.



This is precisely the reason why `getIssues` at Line#146 `Cannot find declaration to go to` and at Line#153, `getIssues` shows as `no Usages found`. (These are the log messages in my editor)

Reply:
Yes @shikhar-scs I got your point. Here a solution could be just moving the function expression ```const getIssues = ...``` above ```const asyncFetch =...```.

Reply:
> Here a solution could be just moving the function expression const getIssues = ... above const asyncFetch =....

Yep @shubhshrma you should change it to `function`

Reply:
> I would still suggest to follow the conventional way as after you a lot of people would be re-reading this code

This is arguable but let's go with what respec has used everywhere, function declarations.

Reply:
Yes please

Reply:
> This is precisely the reason why getIssues at Line#146 Cannot find declaration to go to and at Line#153, getIssues shows as no Usages found. (These are the log messages in my editor)

Just to confirm these are log messages in your editor or browser? Because the PR is passing all tests and locally the issue referencing respec feature is working normal.

Reply:
No. No. No. I use Webstorm IDE. It always auto-suggests changes and any minor errors in tooltips on hover. Similar to these. 
> This is precisely the reason why getIssues at Line#146 Cannot find declaration to go to and at Line#153, getIssues shows as no Usages found. (These are the log messages in my editor)

Reply:
> I use Webstorm IDE

Got it.